UI/Pages/Enquiry/Respond/CheckYourAnswers.cshtml.cs

```diff
@@ -61,24 +63,45 @@ public async Task<IActionResult> OnPostAsync()
         var isValidMagicLink =
             await _mediator.Send(new IsValidMagicLinkTokenQuery(Data.Token, Data.SupportReferenceNumber, Data.TuitionPartnerSeoUrl, true));
 
-        if (!isValidMagicLink)
+        ResponseConfirmationModel? responseConfirmationModel;
+
+        var enquiryResponseConfirmationModelKey = $"{EnquiryResponseConfirmationModelKey}-{Data.SupportReferenceNumber}";
+        var enquiryResponseFormPostTimestampKey = $"{EnquiryResponseFormPostTimestampKey}-{Data.SupportReferenceNumber}";
+
+        var isDuplicateFormPost = await _sessionService.IsDuplicateFormPostAsync(enquiryResponseFormPostTimestampKey);
+
+        if (!isDuplicateFormPost)
         {
-            return NotFound();
-        }
+            if (!isValidMagicLink)
```

Comment:
To avoid unnecessary calls to isDuplicateFormPost, we ought to extract the validation logic for the magic link from !isDuplicateFormPost. This would enable us to return "not found" immediately when an invalid magic link is encountered.